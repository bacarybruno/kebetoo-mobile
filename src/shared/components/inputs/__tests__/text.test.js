import { TextInput as RNTextInput } from 'react-native'
import { fireEvent } from 'react-native-testing-library'
import { act } from 'react-test-renderer'

import setupTest from '@app/config/jest-setup'

import TextInput, { PopoverTooltip } from '../text'

const givenTextInput = setupTest(TextInput)({
  onValueChange: jest.fn(),
  onBlur: jest.fn(),
  fieldName: 'fullname',
})

it('renders TextInput', () => {
  const { wrapper } = givenTextInput()
  expect(wrapper.toJSON()).toMatchSnapshot()
})

it('shows popover tooltip on error', () => {
  const { wrapper } = givenTextInput({ error: 'An error occured' })
  expect(wrapper.root.findAllByType(PopoverTooltip).length).toBe(1)
})

it('handles value change', () => {
  const { wrapper, props } = givenTextInput()
  const message = 'hello world'
  act(() => {
    fireEvent.changeText(wrapper.root.findByType(RNTextInput), message)
  })
  expect(props.onValueChange).toBeCalledTimes(1)
  expect(props.onValueChange).toBeCalledWith(message, props.fieldName)
})

it('handles onBlur', () => {
  const { wrapper, props } = givenTextInput()
  act(() => {
    fireEvent(wrapper.root.findByType(RNTextInput), 'onBlur')
  })
  expect(props.onBlur).toBeCalledTimes(1)
  expect(props.onBlur).toBeCalledWith(props.fieldName)
})
