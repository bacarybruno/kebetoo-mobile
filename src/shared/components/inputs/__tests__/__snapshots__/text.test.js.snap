// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TextInput 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F5F5F5",
        "borderColor": "rgba(229, 229, 234, 255)",
        "borderRadius": 15,
        "borderWidth": 1,
        "flexDirection": "row",
        "minHeight": 48,
        "paddingLeft": 16,
      },
      undefined,
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    blurOnSubmit={false}
    onBlur={[Function]}
    onChangeText={[Function]}
    placeholderTextColor="rgba(174, 174, 178, 255)"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgba(0,0,0,0.87)",
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": -0.32,
          "lineHeight": 21,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value={null}
  />
</View>
`;
